src/common/perf_counters.cc

```diff
@@ -50,7 +50,10 @@ void PerfCountersCollectionImpl::add(PerfCounters *l)
   for (unsigned int i = 0; i < l->m_data.size(); ++i) {
     PerfCounters::perf_counter_data_any_d &data = l->m_data[i];
 
-    std::string path = l->get_name();
+    // We get the name of the counter with key_name because 
+    // labeled perfcounters have null characters in between
+    // which can cause problems when appending the data name.
```

Comment:
And what would happen when two `PerfCounters` instances that have the same "base" name and differ only in the value of one particular label are added to the collection?

This "problematic" naming scheme is actually convenient because the labels are part of the name and are carried along with the "base" name automatically.  I would look in the direction of making `by_path` work with labels (e.g. by changing it to be indexed by a pair of strings instead of a single string or by making it a two-level map).

Reply:
Also, if `PerfCountersCollectionImpl::add` is getting adjusted, `PerfCountersCollectionImpl::remove` and possibly other things that operate on `by_path` need to be adjusted accordingly.